### PR TITLE
feat(java/python/golang): Implement the magicNumber feature

### PR DESCRIPTION
**Pull request by LiangliangSui**
## What does this PR do?

This PR implements the MagicNumer feature in [xlang_serialization_spec.md](https://github.com/apache/incubator-fury/blob/main/docs/specification/xlang_serialization_spec.md#fury-header).

## Does this PR introduce any user-facing change?

- [ ] Does this PR introduce any public API change?
- [ ] Does this PR introduce any binary protocol compatibility change?

### CODE REVIEW

**Comment by LiangliangSui** [comment]
Hi @chaokunyang , this pr is ready for review, could you help review this?

File: python/pyfury/_serialization.pyx [hunk 1]
```diff
@@ -1058,6 +1061,11 @@ cdef class Fury:
             self.unpickler = Unpickler(buffer)
         if unsupported_objects is not None:
             self._unsupported_objects = iter(unsupported_objects)
+        if self.language == Language.XLANG:
+            magic_numer = buffer.read_int16()
+            assert magic_numer == MAGIC_NUMBER, (
+                "The XLANG cross-language protocol is not currently used. For details, see "
```
Comment (chaokunyang) [inline python/pyfury/_serialization.pyx:1067]: Clang and cross language seems a little redundant. Other parts look good to me. But could we refine the error message？

Comment (LiangliangSui) [reply python/pyfury/_serialization.pyx:1067]: How about the error msg below?
`The xlang serialization specification must start with 0x62d4. Please confirm whether the serialization is based on the xlang protocol. For details, see https://github.com/apache/incubator-fury/blob/main/docs/specification/xlang_serialization_spec.md`

Comment (chaokunyang) [reply python/pyfury/_serialization.pyx:1067]: how about `The fury xlang serialization must start with magic number 0x62d4. Please check whether the serialization is based on the xlang protocol and the data didn't corrupt.`

Comment (LiangliangSui) [reply python/pyfury/_serialization.pyx]: Great! updated.

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**